### PR TITLE
(SIMP-4900) First cut at an x2go module

### PR DESCRIPTION
**Pull request by trevor-vaughan**
SIMP-4900 #close

### CODE REVIEW

File: data/common.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+x2go::server::agent_options:
```
Comment (jeefberkey) [inline data/common.yaml]: There should be a `lookup_options` setting for each hash (unless you didn't want to use merging)

File: manifests/server.pp [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+#
+# @param config_file
+#   The location of the main server configuration file
+#
+# @param agent_config_file
+#   The location of the agent configuration file
+#
+# @author https://github.com/simp/pupmod-simp-x2go/graphs/contributors
+#
+class x2go::server (
```
Comment (jeefberkey) [inline manifests/server.pp]: Isn't there a service that needs to be set up here? `x2gocleansessions`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+safe defaults in place.
+
+## Usage
+
+The ``x2go`` client is installed by default. To disable this, set
+``x2go::client`` to ``false`` in Hiera.
+
+To install and configure the ``x2go`` server, set ``x2go::server`` to ``true`` in Hiera.
+
+``x2go`` requires a functioning window manager on the server to be useful and
```
Comment (jeefberkey) [inline README.md]: Some good docs on that from the project: https://wiki.x2go.org/doku.php/doc:de-compat

**Comment by lnemsick-simp** [comment]
I tried this module along with the updated gdm and gnome modules on  a workstation .  It largely worked (after I enabled the epel repo to get the mate-related RPMs.).  However, there is one odd behavior that I haven't tracked down:  When I right click and select 'Open in terminal', nothing happens.  I have to go to the full drop-down menu to start a terminal.

**Comment by lnemsick-simp** [comment]
Are we going to add MATE-related RPMs to the ISO?

**Comment by m-morrone** [comment]
The right click issue reported by lnesmick is an environment issue.  By default, no terminal emulator is set in MATE so nothing launches.  To set a terminal emulator, click on system at the top of the screen, go down to preferences > and select preferred applications (4 from the bottom).  click on the system tab and select terminal from the list for terminal emulator.

**Comment by trevor-vaughan** [comment]
@lnemsick-simp Updated this to 'alpha' status so that we can merge it for testing and split out the code base a bit more. This is an important historical snapshot to keep

**Pull request merged**